improver_tests/fire_weather/test_fine_fuel_moisture_content.py

```
@@ -0,0 +1,619 @@
+            np.array([70, 70, 70, 70, 70]),
+            np.array([49.31, 49.31, 49.31, 49.31, 49.31]),
+        ),
+        # Case 3: Mixed moisture content values
+        (
+            np.array([10, 70, 30, 80, 50]),
+            np.array([90.84, 49.31, 73.87, 44.52, 60.34]),
+        ),
+        # Case 4: Near maximum moisture content (250)
+        (
+            np.array([250, 250, 250, 250, 250]),
+            np.array([0.0, 0.0, 0.0, 0.0, 0.0]),
+        ),
+    ],
+)
+def test__calculate_ffmc_from_moisture_content(
+    moisture_content: np.ndarray,
+    expected_output: np.ndarray,
+) -> None:
+    """Test _calculate_ffmc_from_moisture_content with various moisture scenarios.
+
+    Verifies FFMC calculation from moisture content.
+
+    Args:
+        moisture_content (np.ndarray): Moisture content values for all grid points.
+        expected_output (np.ndarray): Expected FFMC output values.
+    """
+    plugin = FineFuelMoistureContent()
+    plugin.moisture_content = moisture_content.copy()
+    ffmc = plugin._calculate_ffmc_from_moisture_content()
+    # Check output type and shape
+    assert isinstance(ffmc, np.ndarray)
+    assert ffmc.shape == moisture_content.shape
+    # Check that FFMC matches expected output
+    assert np.allclose(ffmc, expected_output, atol=0.01)
+
+
+@pytest.mark.parametrize(
+    "temp_val, precip_val, rh_val, wind_val, ffmc_val, expected_output",
+    [
+        # Case 0: Typical mid-range values
+        (20.0, 1.0, 50.0, 10.0, 85.0, 83.85),
+        # Case 1: All zeros (edge case)
+        (0.0, 0.0, 0.0, 0.0, 0.0, 22.30),
+        # Case 2: High temp, no precip, low RH, high wind (produces high output FFMC)
+        (35.0, 0.0, 15.0, 25.0, 90.0, 96.75),
+        # Case 3: Low temp, high precip, high RH, low wind (produces low output FFMC)
+        (10.0, 15.0, 95.0, 5.0, 85.0, 20.70),
+        # Case 4: Precipitation just below threshold (should not adjust)
+        (20.0, 0.4, 50.0, 10.0, 85.0, 86.82),
+    ],
+)
```

I think there should be an additional test that mirrors this one, except that it tests that incorrect/bad input values (e.g. negative precipitation rate or humidity, very low temperature, etc.) raise an error, and don't silently return some meaningless output.